Tests failed on windows

When running `pytest fsspec` on `windows` 4 tests are failing.
```python
========================================================================================================= test session starts ==========================================================================================================
platform win32 -- Python 3.8.5, pytest-6.1.1, py-1.9.0, pluggy-0.13.1
rootdir: C:\...\filesystem_spec
collected 341 items / 8 skipped / 333 selected

fsspec\implementations\tests\test_cached.py .sssss....s..F...............................                                                                                                                                         [ 13%]
fsspec\implementations\tests\test_common.py ..                                                                                                                                                                                    [ 13%]
fsspec\implementations\tests\test_dask.py .                                                                                                                                                                                       [ 14%]
fsspec\implementations\tests\test_ftp.py ssssssssss                                                                                                                                                                               [ 17%]
fsspec\implementations\tests\test_hdfs.py sssss                                                                                                                                                                                   [ 18%]
fsspec\implementations\tests\test_local.py ...................................xx..........x..F...                                                                                                                                 [ 34%]
fsspec\implementations\tests\test_memory.py .......                                                                                                                                                                               [ 36%]
fsspec\implementations\tests\test_zip.py ...........                                                                                                                                                                              [ 39%]
fsspec\tests\test_api.py ............x....                                                                                                                                                                                        [ 44%]
fsspec\tests\test_async.py .                                                                                                                                                                                                      [ 44%]
fsspec\tests\test_caches.py ............................................................................                                                                                                                          [ 67%]
fsspec\tests\test_compression.py ..sss                                                                                                                                                                                            [ 68%]
fsspec\tests\test_core.py .........FF....s..                                                                                                                                                                                      [ 73%]
fsspec\tests\test_file.py ssssssssss                                                                                                                                                                                              [ 76%]
fsspec\tests\test_mapping.py ......                                                                                                                                                                                               [ 78%]
fsspec\tests\test_registry.py ss....                                                                                                                                                                                              [ 80%]
fsspec\tests\test_spec.py .............................ssssssssss                                                                                                                                                                 [ 91%]
fsspec\tests\test_utils.py ............................                                                                                                                                                                           [100%]

=============================================================================================================== FAILURES ===============================================================================================================
________________________________________________________________________________________ test_local_filecache_creates_dir_if_needed[blockcache] ________________________________________________________________________________________

impl = 'blockcache'

    @pytest.mark.parametrize("impl", ["filecache", "simplecache", "blockcache"])
    def test_local_filecache_creates_dir_if_needed(impl):
        import tempfile

        original_location = tempfile.mkdtemp()
        cache_location = tempfile.mkdtemp()
        os.rmdir(cache_location)
        assert not os.path.exists(cache_location)

        try:
            original_file = os.path.join(original_location, "afile")
            data = b"test data"
            with open(original_file, "wb") as f:
                f.write(data)

            # we can access the file and read it
            fs = fsspec.filesystem(
                impl, target_protocol="file", cache_storage=cache_location
            )

            with fs.open(original_file, "rb") as f:
                data_in_cache = f.read()

            assert os.path.exists(cache_location)

        finally:
>           shutil.rmtree(cache_location)

fsspec\implementations\tests\test_cached.py:243:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
..\..\..\Anaconda3\envs\pd\lib\shutil.py:737: in rmtree
    return _rmtree_unsafe(path, onerror)
..\..\..\Anaconda3\envs\pd\lib\shutil.py:615: in _rmtree_unsafe
    onerror(os.unlink, fullname, sys.exc_info())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

path = 'C:\\...\\AppData\\Local\\Temp\\tmpovmjs47x', onerror = <function rmtree.<locals>.onerror at 0x000002531746D3A0>

    def _rmtree_unsafe(path, onerror):
        try:
            with os.scandir(path) as scandir_it:
                entries = list(scandir_it)
        except OSError:
            onerror(os.scandir, path, sys.exc_info())
            entries = []
        for entry in entries:
            fullname = entry.path
            if _rmtree_isdir(entry):
                try:
                    if entry.is_symlink():
                        # This can only happen if someone replaces
                        # a directory with a symlink after the call to
                        # os.scandir or entry.is_dir above.
                        raise OSError("Cannot call rmtree on a symbolic link")
                except OSError:
                    onerror(os.path.islink, fullname, sys.exc_info())
                    continue
                _rmtree_unsafe(fullname, onerror)
            else:
                try:
>                   os.unlink(fullname)
E                   PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\...\AppData\\Local\\Temp\\tmpovmjs47x\\acb8cce76f53b46064484769e0682fde03f34d54c6a8b79fc0f9f4fa462e234f'

..\..\..\Anaconda3\envs\pd\lib\shutil.py:613: PermissionError
____________________________________________________________________________________________________ test_strip_protocol_expanduser ____________________________________________________________________________________________________

    def test_strip_protocol_expanduser():
        path = "file://~\\foo\\bar" if sys.platform == "win32" else "file://~/foo/bar"
        stripped = LocalFileSystem._strip_protocol(path)
        assert path != stripped
        assert "file://" not in stripped
>       assert stripped.startswith(os.path.expanduser("~"))
E       AssertionError: assert False
E        +  where False = <built-in method startswith of str object at 0x0000025319DFEF30>('C:\\...')
E        +    where <built-in method startswith of str object at 0x0000025319DFEF30> = 'C:/.../foo/bar'.startswith
E        +    and   'C:\\....' = <function expanduser at 0x00000253146723A0>('~')
E        +      where <function expanduser at 0x00000253146723A0> = <module 'ntpath' from 'C:\\Anaconda3\\envs\\pd\\lib\\ntpath.py'>.expanduser
E        +        where <module 'ntpath' from 'C:\\Anaconda3\\envs\\pd\\lib\\ntpath.py'> = os.path

fsspec\implementations\tests\test_local.py:514: AssertionError
______________________________________________________________________________________________________________ test_list _______________________________________________________________________________________________________________

    def test_list():
        here = os.path.abspath(os.path.dirname(__file__))
        flist = os.listdir(here)
        plist = [os.path.join(here, p) for p in flist]
        of = open_files(plist)
        assert len(of) == len(flist)
>       assert [f.path for f in of] == plist
E       AssertionError: assert ['C:/.../...file.py', ...] == ['C:\\...\\...file.py', ...]
E         At index 0 diff: 'C:/.../filesystem_spec/fsspec/tests/test_api.py' != 'C:\\...\\filesystem_spec\\fsspec\\tests\\test_api.py'
E         Use -v to get the full diff

fsspec\tests\test_core.py:82: AssertionError
___________________________________________________________________________________________________________ test_pathobject ____________________________________________________________________________________________________________

tmpdir = 'C:\\...\\AppData\\Local\\Temp\\pytest-of-...\\pytest-17\\test_pathobject0'

    def test_pathobject(tmpdir):
        import pathlib

        tmpdir = str(tmpdir)
        plist_str = [os.path.join(str(tmpdir), f) for f in ["a", "b"]]
        open(plist_str[0], "w").write("first file")
        open(plist_str[1], "w").write("second file")
        plist = [pathlib.Path(p) for p in plist_str]
        of = open_files(plist)
        assert len(of) == 2
>       assert [f.path for f in of] == plist_str
E       AssertionError: assert ['C:/...athobject0/b'] == ['C:\\...thobject0\\b']
E         At index 0 diff: 'C:/.../AppData/Local/Temp/pytest-of-.../pytest-17/test_pathobject0/a' != 'C:\\...\\AppData\\Local\\Temp\\pytest-of-...\\pytest-17\\test_pathobject0\\a'
E         Use -v to get the full diff

fsspec\tests\test_core.py:95: AssertionError
======================================================================================================= short test summary info ========================================================================================================
FAILED fsspec/implementations/tests/test_cached.py::test_local_filecache_creates_dir_if_needed[blockcache] - PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\...
FAILED fsspec/implementations/tests/test_local.py::test_strip_protocol_expanduser - AssertionError: assert False
FAILED fsspec/tests/test_core.py::test_list - AssertionError: assert ['C:/...file.py', ...] == ['C:\\...file.py', ...]
FAILED fsspec/tests/test_core.py::test_pathobject - AssertionError: assert ['C:/...athobject0/b'] == ['C:\\...thobject0\\b']
======================================================================================== 4 failed, 286 passed, 55 skipped, 4 xfailed in 18.34s =========================================================================================
```

--- a/fsspec/implementations/tests/test_cached.py
+++ b/fsspec/implementations/tests/test_cached.py
@@ -223,24 +223,20 @@ def test_local_filecache_creates_dir_if_needed(impl):
     os.rmdir(cache_location)
     assert not os.path.exists(cache_location)
 
-    try:
-        original_file = os.path.join(original_location, "afile")
-        data = b"test data"
-        with open(original_file, "wb") as f:
-            f.write(data)
+    original_file = os.path.join(original_location, "afile")
+    data = b"test data"
+    with open(original_file, "wb") as f:
+        f.write(data)
 
-        # we can access the file and read it
-        fs = fsspec.filesystem(
-            impl, target_protocol="file", cache_storage=cache_location
-        )
+    # we can access the file and read it
+    fs = fsspec.filesystem(
+        impl, target_protocol="file", cache_storage=cache_location
+    )
 
-        with fs.open(original_file, "rb") as f:
-            data_in_cache = f.read()
+    with fs.open(original_file, "rb") as f:
+        data_in_cache = f.read()
 
-        assert os.path.exists(cache_location)
-
-    finally:
-        shutil.rmtree(cache_location)
+    assert os.path.exists(cache_location)
 
     assert data_in_cache == data
 

--- a/fsspec/implementations/tests/test_cached.py
+++ b/fsspec/implementations/tests/test_cached.py
@@ -229,9 +229,7 @@ def test_local_filecache_creates_dir_if_needed(impl):
         f.write(data)
 
     # we can access the file and read it
-    fs = fsspec.filesystem(
-        impl, target_protocol="file", cache_storage=cache_location
-    )
+    fs = fsspec.filesystem(impl, target_protocol="file", cache_storage=cache_location)
 
     with fs.open(original_file, "rb") as f:
         data_in_cache = f.read()

--- a/fsspec/implementations/tests/test_local.py
+++ b/fsspec/implementations/tests/test_local.py
@@ -511,7 +511,7 @@ def test_strip_protocol_expanduser():
     stripped = LocalFileSystem._strip_protocol(path)
     assert path != stripped
     assert "file://" not in stripped
-    assert stripped.startswith(os.path.expanduser("~"))
+    assert stripped.startswith(os.path.expanduser("~").replace("\\", "/"))
 
 
 def test_iterable(tmpdir):

--- a/fsspec/tests/test_core.py
+++ b/fsspec/tests/test_core.py
@@ -76,7 +76,7 @@ def test_xz_lzma_compressions():
 def test_list():
     here = os.path.abspath(os.path.dirname(__file__))
     flist = os.listdir(here)
-    plist = [os.path.join(here, p) for p in flist]
+    plist = [os.path.join(here, p).replace("\\", "/") for p in flist]
     of = open_files(plist)
     assert len(of) == len(flist)
     assert [f.path for f in of] == plist
@@ -86,7 +86,7 @@ def test_pathobject(tmpdir):
     import pathlib
 
     tmpdir = str(tmpdir)
-    plist_str = [os.path.join(str(tmpdir), f) for f in ["a", "b"]]
+    plist_str = [os.path.join(str(tmpdir), f).replace("\\", "/") for f in ["a", "b"]]
     open(plist_str[0], "w").write("first file")
     open(plist_str[1], "w").write("second file")
     plist = [pathlib.Path(p) for p in plist_str]

